public_html/wp-content/mu-plugins/blocks/assets/src/organizers/organizers-select.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * External dependencies
+ */
+import { get, includes } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+const { Dashicon }       = wp.components;
+const { Component }      = wp.element;
+const { decodeEntities } = wp.htmlEntities;
+const { __ }             = wp.i18n;
+
+/**
+ * Internal dependencies
+ */
+import AvatarImage from '../shared/avatar';
+import ItemSelect  from '../shared/item-select';
+
+class OrganizersSelect extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			wcb_organizer      : [],
+			wcb_organizer_team : [],
+			loading            : true,
+		};
+
+		this.buildSelectOptions = this.buildSelectOptions.bind( this );
+	}
+
+	componentWillMount() {
+		this.isStillMounted = true;
+
+		const { allOrganizerPosts, allOrganizerTerms } = this.props;
+
+		const parsedPosts = allOrganizerPosts.then(
+			( fetchedPosts ) => {
+				const posts = fetchedPosts.map( ( post ) => {
+					return {
+						label  : decodeEntities( post.title.rendered.trim() ) || __( '(Untitled)', 'wordcamporg' ),
+						value  : post.id,
+						type   : 'wcb_organizer',
+						avatar : post.avatar_urls[ '24' ],
+					};
+				} );
+
+				if ( this.isStillMounted ) {
+					this.setState( { wcb_organizer: posts } );
+				}
+			}
+		);
+
+		const parsedTerms = allOrganizerTerms.then(
+			( fetchedTerms ) => {
+				const terms = fetchedTerms.map( ( term ) => {
+					return {
+						label : decodeEntities( term.name ) || __( '(Untitled)', 'wordcamporg' ),
+						value : term.id,
+						type  : 'wcb_organizer_team',
+						count : term.count,
+					};
+				} );
+
+				if ( this.isStillMounted ) {
+					this.setState( { wcb_organizer_team: terms } );
+				}
+			}
+		);
+
+		Promise.all( [ parsedPosts, parsedTerms ] ).then( () => {
+			this.setState( { loading: false } );
+		} );
+	}
+
+	componentWillUnmount() {
+		this.isStillMounted = false;
+	}
+
+	buildSelectOptions( mode ) {
+		const options = [];
+
+		const labels = {
+			wcb_organizer      : __( 'Organizers', 'wordcamporg' ),
+			wcb_organizer_team : __( 'Teams',      'wordcamporg' ),
+		};
+
+		for ( const type in this.state ) {
+			if ( ! this.state[ type ].length ) {
```

Comment:
PhpStorm gives me an inspection warning here because there's not a `hasOwnProperty` check. The purpose of that check is to make sure you're only iterating over properties in `this.state` that you have explicitly set, and not ones that are inherited from the prototype. I'm not sure that's actually a big deal here, but you can see how I mitigated the inspection warning in **sessions-select.js**.

Reply:
Yeah, IIRC I intentionally left that out, because it felt like it added clutter without a practical benefit, since in this case there shouldn't be any enumerable properties coming from the prototype (and `for...in` ignores non-enumerable ones).

I tried using `for ( const type in Object.getOwnPropertyNames( this.state ) )` for a cleaner approach, but it didn't work for some reason. It might if I play with it a bit more, though, I might have just being making a basic mistake.

I kinda feel like that inspection is over-zealous, **but** I also feel like consistency is important, and triggering inspections hurts devex by distracting us from the code, so in this case it's probably better to add the `hasOwnProperty()` check if `getOwnPropertyNames()` won't work.

_Side note: It might be nice to share some basic PhpStorm config for things like inspections, but it seems like [PhpStorm makes that way more trouble than it's worth](https://www.reddit.com/r/PHP/comments/al87zr/how_do_you_guys_handle_versioning_of_the_phpstorm/)._

Reply:
Ah, yeah, I just need to use `Object.getOwnPropertyDescriptors()` instead :)

It doesn't look like IE11 supports that. Babel might transpile it into something more primitive, or provide a polyfill or something. I'll have to setup a VM to test.

Reply:
Yup, works in IE11. Fixed in 2809c1e4b080c8e8798f952f012d736b4ee5aa3a.